tokens/usecase/pricing/coingecko/pricing_coingecko.go

```diff
@@ -0,0 +1,132 @@
+package coingeckopricing
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/cache"
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	cacheHitsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "sqs_pricing_cache_hits_total",
+			Help: "Total number of pricing cache hits",
+		},
+		[]string{"base", "quote"},
+	)
+	cacheMissesCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "sqs_pricing_cache_misses_total",
+			Help: "Total number of pricing cache misses",
+		},
+		[]string{"base", "quote"},
+	)
+)
+
+type coingeckoPricing struct {
+	TUsecase      mvc.TokensUsecase
+	cache         *cache.Cache
+	cacheExpiryNs time.Duration
+	quoteCurrency string
+	coingeckoUrl  string
+}
+
+func init() {
+	prometheus.Register(cacheHitsCounter)
+	prometheus.Register(cacheMissesCounter)
+}
+
+func New(routerUseCase mvc.RouterUsecase, tokenUseCase mvc.TokensUsecase, config domain.PricingConfig) domain.PricingSource {
+	return &coingeckoPricing{
+		TUsecase:      tokenUseCase,
+		cache:         cache.New(),
+		cacheExpiryNs: time.Duration(config.CacheExpiryMs) * time.Millisecond,
+		quoteCurrency: config.CoingeckoQuoteCurrency,
+		coingeckoUrl:  config.CoingeckoUrl,
+	}
+}
+
+// GetPrice implements pricing.PricingStrategy.
+// quoteDenom is ignored as it uses always coingecko-quote-currency in config.json
+func (c *coingeckoPricing) GetPrice(ctx context.Context, baseDenom string, quoteDenom string, opts ...domain.PricingOption) (osmomath.BigDec, error) {
```

Comment:
Could we add a functional style test into CI that is controlled by an environment variable to find all listed tokens as defined by [mainnetState.Tokens](https://github.com/osmosis-labs/sqs/blob/db614d1d38a3d02661a41312866557748ff89a3c/tokens/usecase/pricing/worker/pricing_worker_test.go#L161) that are unsupported in coingecko pricing?

Please see the following example:
https://github.com/osmosis-labs/sqs/blob/db614d1d38a3d02661a41312866557748ff89a3c/tokens/usecase/pricing/worker/pricing_worker_test.go#L146-L150